lib/src/main/java/com/diffplug/spotless/markdown/FreshMarkStep.java

```diff
@@ -52,13 +51,21 @@ private FreshMarkStep() {}
 	private static final String FORMATTER_CLASS = "com.diffplug.freshmark.FreshMark";
 	private static final String FORMATTER_METHOD = "compile";
 
+	private final JarState.Promised jarState;
+	private final Map<String, ?> properties;
+
+	private FreshMarkStep(JarState.Promised jarState, Map<String, ?> properties) {
+		this.jarState = jarState;
+		this.properties = properties;
+	}
+
 	/** Creates a formatter step for the given version and settings file. */
-	public static FormatterStep create(Supplier<Map<String, ?>> properties, Provisioner provisioner) {
+	public static FormatterStep create(Map<String, ?> properties, Provisioner provisioner) {
 		return create(defaultVersion(), properties, provisioner);
 	}
 
 	/** Creates a formatter step for the given version and settings file. */
-	public static FormatterStep create(String version, Supplier<Map<String, ?>> properties, Provisioner provisioner) {
+	public static FormatterStep create(String version, Map<String, ?> properties, Provisioner provisioner) {
```

Comment:
I'm removing the `Supplier` wrapper for properties, is there any reason to keep it? Might it be related to the failure?

Reply:
Not sure why it was a `Supplier`. I'm a big hesitant to break this public API for no reason, but the `Supplier` is a strange choice. No idea why it would be related to the failure.

Reply:
I removed it cause `properties.get()` would throw `Exception`, but `FormatterStep create` didn't, which means many invocations need to throw exceptions too.